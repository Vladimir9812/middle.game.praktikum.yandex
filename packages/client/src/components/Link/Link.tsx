import { Link as ReactRouterLink } from 'react-router-dom';
import { Link as ChakraLink } from '@chakra-ui/react';
import { ReactElement } from 'react';

type Properties = {
  [x: string]: any;
};
export function Link(properties: Properties): ReactElement<Properties> {
  // eslint-disable-next-line react/jsx-props-no-spreading
  return <ChakraLink as={ReactRouterLink} {...properties} />;
}
