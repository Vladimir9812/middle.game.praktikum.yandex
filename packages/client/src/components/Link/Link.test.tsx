import { MemoryRouter } from 'react-router-dom';
// eslint-disable-next-line import/no-extraneous-dependencies
import ReactTestRenderer from 'react-test-renderer';

import { Link } from './Link';
// const ReactTestRenderer = require('react-test-renderer');

it('hello jest', () => {
  const tree = ReactTestRenderer.create(
    <MemoryRouter>
      <Link to="/forum/3">3</Link>
    </MemoryRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
