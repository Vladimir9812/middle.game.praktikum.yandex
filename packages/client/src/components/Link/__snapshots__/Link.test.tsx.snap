// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello jest 1`] = `
<a
  className="chakra-link css-0"
  href="/forum/3"
  onClick={[Function]}
>
  3
</a>
`;
